source/thunkify.js

```diff
@@ -0,0 +1,29 @@
+import _curry1 from './internal/_curry1';
+import _arity from './internal/_arity';
+
+/**
+ * Creates a thunk out of a function. A thunk delays a calculation until
+ * its result is needed, providing lazy evaluation of arguments.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig (a -> b) -> (a -> b) // FIXME: write appropiate signature
```

Comment:
I am opening the discussion on a type sig here. Something like 

`((a, b, ..., j) -> k) -> (a, b, ..., j) -> (() -> k)`

would probably do the trick. I haven't gone back to look at the standard way to notate variadic functions though...

Reply:
I went for something similar first time writing it, based on how `flip` [signature was written](https://github.com/ramda/ramda/blob/master/source/flip.js#L13). But desisted since I wasn't quite sure if `()` is a valid notation (isn't it just a scalar value?)

Reply:
This would be the "aspirational" signature: `(a -> b) -> a -> () -> b`

capturing the variadic-ness of `fn` makes it tricky to right a coherent sig.

`(a... -> b) -> a... -> () -> b`

?

Reply:
@buzzdecafe Based on how previous variadic function's signatures were written, I'd say @Bradcomp is on the right track here (i.e.: `(a, b..., j)`).

So, merging both suggestions, you'll end up with his proposal.

Reply:
I like @Bradcomp's suggestion here.  At some point, I want to switch all internally variadic signatures to an ellipses-terminated list of signatures, but Brad's version is consistent for now with our current variadic signatures.

Reply:
Changing it to match @Bradcomp's sig, then.